cli/common.h

```diff
@@ -98,6 +98,12 @@
 #define TOSTRING(x) #x
 #define STRINGIFY(x) TOSTRING(x)
 
+// Function attribute: this macro is used to bypass Windows build
+// failing since it doesn't recognize the attribute.
+#if defined( WIN32 ) || defined( _WIN32 ) || defined( __WIN32__ )
+  #define __attribute__(X)
+#endif
+
```

Comment:
We should avoid "quick and dirty workaround" at all cost. Also if someone using `mingw` on windows it'll override the macro (with a warning) because mingw has this `__attribute__` feature (which is something compiler dependent and not a platform thing). The proper way to make a portable deprication macro is (I believe)
```c
// Reference: https://stackoverflow.com/a/22823913/10846399
#ifdef __GNUC__
    #define DEPRECATED(x) x __attribute__((deprecated))
#elif defined _MSC_VER
    #define DEPRECATED(x) __pragma deprecated(x)
#else
    #define DEPRECATED(x) x
#endif
```

Reply:
Thank you! We will come back to this in the near future.